server/build.gradle

```diff
@@ -40,9 +40,13 @@ dependencies {
     // jCommander
     optionalImplementation libs.jcommander
 
-    // jGit
-    implementation libs.jgit
-    implementation libs.jgit.ssh.jsch
+    // Hacky way to use different jGit versions depending on the test Java version.
+    // We will use jgit6 after we drop the support for Java 8 client.
+    if (project.ext.testJavaVersion >= 11) {
```

Comment:
By the way, I thought, an info message for the current JGit version would be useful.
```java
static {
    InputStream is = null;
    try {
        is = SystemReader.class.getClassLoader().getResourceAsStream("META-INF/maven/org.eclipse.jgit/org.eclipse.jgit/pom.properties");
        final Properties props = new Properties();
        props.load(is);
        final Object jgitVersion = props.get("version");
        if (jgitVersion != null) {
            logger.info("Using JGit: {}", jgitVersion);
        }
    } catch (IOException e) {
        logger.debug("Failed to read JGit version", e);
    } finally {
        if (is != null) {
            try {
                is.close();
            } catch (IOException e) {
                // ignore
            }
        }
    }
}
```
Let me send a PR for that.

Reply:
> Yes, it works.

Does it work when we specify `-PtestJavaVersion=8`? 🤔

Reply:
- When `-PtestJavaVersion=8` was set, JGit 5 was chosen.
```sh
❯ gw clean :client:java-armeria:test -PtestJavaVersion=8 
❯ ag "Using JGit"  
java-armeria/build/test-results/test/TEST-com.linecorp.centraldogma.client.armeria.ArmeriaCentralDogmaTest.xml
57:2023-08-23 21:01:05.980 [INFO ](c.l.c.s.i.s.r.g.GitRepository) [repository-worker-6-1] Using JGit: 5.13.0.202109080827-r
```
- When `-PtestJavaVersion=8 -PminimumJavaVersion=11` is set, no tests are run since `testJavaVersion` is lower than `targerJavaVersion`. `minimumJavaVersion` raises `targerJavaVersion` to 11.
```sh
❯ gw clean :client:java-armeria:test -PtestJavaVersion=8 -PminimumJavaVersion=11

Skipping tests for :client:java-armeria since the testJavaVersion(8) is smaller than the targetJavaVersion(11)
```
- If we set `PtestJavaVersion` to 11, the client tests can be run with JGit 6.
```sh
~/src/centraldogma/client log-jgit-version*
❯ gw clean :client:java-armeria:test -PtestJavaVersion=11 -PminimumJavaVersion=11

java-armeria/build/reports/tests/test/classes/com.linecorp.centraldogma.client.armeria.ArmeriaCentralDogmaTest.html
146:2023-08-23 21:07:18.011 [INFO ](c.l.c.s.i.s.r.g.GitRepository) [repository-worker-6-1] Using JGit: 6.6.0.202305301015-r
```

This is the change used to test.
```diff
diff --git server/build.gradle server/build.gradle
index 7f828d61..a2b0f2b1 100644
--- server/build.gradle
+++ server/build.gradle
@@ -42,7 +42,7 @@ dependencies {
 
     // Hacky way to use different jGit versions depending on the test Java version.
     // We will use jgit6 after we drop the support for Java 8 client.
-    if (project.ext.testJavaVersion >= 11) {
+    if (project.ext.targetJavaVersion >= 11) {
         implementation libs.jgit6
     } else {
         implementation libs.jgit
```

Reply:
Anyway, it's too late so let's keep discussing tomorrow. 😉

Reply:
We don’t need to set an extra properties since we publish the server module with JGit 5.
JGit 6 will be chosen only mininumJavaVersion is higher than 11.

When packing a tarball, JGit 5 is excluded and JGit is added instead.

Reply:
Ah ha, now I understand. Well actually, publishing the server module with JGit 6 was my intention but I guess I was wrong because users would still need the server module for testing. Let me make a change, then. Thanks!

Reply:
The changed is pushed: https://github.com/line/centraldogma/pull/830/commits/51184221667e659c21f009cf0abae90b9e2363b0
PTAL. 🙇

Reply:
It looks 👍!